bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BTupleType.java

```diff
@@ -30,21 +30,28 @@
 public class BTupleType extends BType {
 
     private List<BType> tupleTypes;
+    private BType restType;
 
     /**
      * Create a {@code BTupleType} which represents the tuple type.
      *
      * @param typeList of the tuple type
+     * @param restType of the tuple type
      */
-    public BTupleType(List<BType> typeList) {
+    public BTupleType(List<BType> typeList, BType restType) {
```

Comment:
Wouldn't it be better to have two constructors instead? We can avoid all the calls passing null when there is not rest descriptor.

Reply:
Fixed in 507ce84